docs/conf.py

```diff
@@ -65,5 +67,50 @@ def extract_summary(doc, document):
 autosummary.extract_summary = extract_summary
 
 
+# Configuration for sphinx.ext.intersphinx
+# https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html
+intersphinx_mapping = {
+    "hail": ("https://hail.is/docs/0.2", None),
+}
+
+
+# sphinx_autodoc_typehints generates references with qualified names.
+# Since Hail re-exports many objects from higher level packages/modules,
+# Hail's documentation does not list all objects by their qualified name.
+# For example, Table is documented as hail.Table, not hail.table.Table.
+# Thus, intersphinx cannot link some of the Hail references generated by
+# sphinx_autodoc_typehints.
+#
+# To work around this, override sphinx_autodoc_typehints's get_annotation_module
+# function and map the qualified names to what Hail's documentation uses.
+original_get_annotation_module = sphinx_autodoc_typehints.get_annotation_module
+
+
+hail_module_map = {
```

Comment:
There might be a more general way of mapping these (like map anything starting with "hail.expr" to "hail.expr") but this covers everything that is currently referenced in gnomad_hail documentation.